lib/project_types/extension/forms/questions/ask_template.rb

```diff
@@ -9,18 +9,24 @@ class AskTemplate
         ]
 
         property! :ctx
+        property :template, accepts: Models::ServerConfig::Development::VALID_TEMPLATES
         property :prompt,
           accepts: ->(prompt) { prompt.respond_to?(:call) },
           default: -> { CLI::UI::Prompt.method(:ask) }
 
         def call(project_details)
           return project_details unless template_required?(project_details)
-          project_details.template = choose_interactively
+          project_details.template = validate_template
```

Comment:
I think I would not extract his into a separate function and simply go with `... = template || choose_interactively`